src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -699,11 +703,6 @@ export default class Fms {
      * @param runwayModel {RunwayModel}
      */
     replaceDepartureProcedure(routeString, runwayModel) {
-        // this is the same procedure that is already set, no need to continue
-        if (this.hasLegWithRouteString(routeString)) {
```

Comment:
Then there's no need to regenerate. But since we don't really have the ability to ask `what expected runway were the waypoints for this model based on?`, we don't really have a way to know whether the runway being assigned is the same as what it was previously. Maybe? : thinking_face: